components/net/http_loader.rs

```diff
@@ -1447,7 +1447,10 @@ fn http_network_fetch(
         &request.method,
         &request.headers,
         request.body.as_mut().and_then(|body| body.take_stream()),
-        request.body.as_ref().map_or(0, |body| body.len()),
+        request
+            .body
+            .as_ref()
+            .map_or(0, |body| body.len().unwrap_or(0)),
```

Comment:
This looks like we'll use a Content-Length of 0 for sending bodies created from a readable stream. That seems like a problem to me.